logstash-plugins/logstash-input-beats
# Introduces expert only event loop thread count setting.

### Description
By default, `netty` event loop creates `available_processors * 2` thread count. This may sometimes lead a resource problem, especially when multiple beats-input pipelines are defined in a single machine.
This PR introduces a configuration to limit a number of threads for netty event loop threads.

### How to locally test
- unfortunately I couldn't find a better way to add a unit test for this feature
- pull the change and run pipeline with following config
- connect, reconnect multiple times the client to the downstream beats-input.
- check the threads (with thread dump tools or ls diagnostics tool) that two (or defined) threads will created for netty event loop named as `nioEventLoopGroup-M-N`

```
input {
    beats {
        codec => json
        port => 5001
        event_loop_threads => 2
        executor_threads => 2
    }
}
output {
    stdout { codec => rubydebug }
}
```

<img width="1493" alt="image" src="https://github.com/logstash-plugins/logstash-input-beats/assets/99575341/cb00e655-4e41-4192-8b90-5d1d1ab5ebe3">
